File: 백준/Silver/10610. 30/30.cc
#include <iostream>
#include <cstring>
#include <algorithm>
using namespace std;

int main() {
    int n;
    int arr[100005];
    char str[100005];
    cin >> str;
    int len = strlen(str);
    for(int i=0;i<len;i++) arr[i] = str[i]-'0';
    sort(arr,arr+len);
    int sum=0;
    for(int i=0;i<len;i++) sum+=arr[i];
    if(arr[0]||sum%3) cout << -1;
    
    else for(int i=len-1;i>=0;i--) cout<<arr[i];
}

File: 백준/Silver/10610. 30/30.cs
using System;
using System.Text;
class Program
{
    static void Main() {
        string s = Console.ReadLine();
        int[] arr = new int[s.Length];
        int cnt=0;
        int len = s.Length;
        StringBuilder sb = new StringBuilder();
        
        foreach(char x in s) arr[cnt++] = x - 48;
        
        Array.Sort(arr);
        int sum=0;
        for(int i=0;i<len;i++) sum+=arr[i];
        if(arr[0] != 0||sum%3 != 0) Console.Write(-1);
        else for(int i=len-1;i>=0;i--) sb.Append(arr[i]);
    
        Console.Write(sb);
    }
}